lib/src/api/method/select.rs

```diff
@@ -97,3 +98,65 @@ where
 		self
 	}
 }
+
+impl<'r, C, R> Select<'r, C, R>
+where
+	C: Connection,
+	R: DeserializeOwned,
+{
+	/// Turns a normal select query into a live query
+	///
+	/// # Examples
+	///
+	/// ```no_run
+	/// # use futures::StreamExt;
+	/// # use surrealdb::opt::Resource;
+	/// # use surrealdb::Result;
+	/// # use surrealdb::Notification;
+	/// # #[derive(Debug, serde::Deserialize)]
+	/// # struct Person;
+	/// #
+	/// # #[tokio::main]
+	/// # async fn main() -> surrealdb::Result<()> {
+	/// # let db = surrealdb::engine::any::connect("mem://").await?;
+	/// #
+	/// // Select the namespace/database to use
+	/// db.use_ns("namespace").use_db("database").await?;
+	///
+	/// // Listen to all updates on a table
+	/// let mut stream = db.select("person").live().await?;
+	/// # let _: Option<Result<Notification<Person>>> = stream.next().await;
+	///
+	/// // Listen to updates on a range of records
+	/// let mut stream = db.select("person").range("jane".."john").live().await?;
+	/// # let _: Option<Result<Notification<Person>>> = stream.next().await;
+	///
+	/// // Listen to updates on a specific record
+	/// let mut stream = db.select(("person", "h5wxrf2ewk8xjxosxtyc")).live().await?;
+	///
+	/// // The returned stream implements `futures::Stream` so we can
+	/// // use it with `futures::StreamExt`, for example.
+	/// while let Some(result) = stream.next().await {
+	///     handle(result);
+	/// }
+	///
+	/// // Handle the result of the live query notification
+	/// fn handle(result: Result<Notification<Person>>) {
+	///     match result {
+	///         Ok(notification) => println!("{notification:?}"),
```

Comment:
Shall we implement `Display`  for `Notification`?

Reply:
I think `Debug` is sufficient and more useful here because we are only printing it for debugging purposes. The caller will mainly care about `notification.data` which contains their deserialised record. We wouldn't have a universally accepted string representation of both `action` and `data` fields anyway. If the caller wants a string representation, it will be trivial for them convert the notification into a string however they see fit.